src/components/molecules/SpaceEditForm/SpaceEditForm.tsx

```diff
@@ -0,0 +1,484 @@
+
+  const roomVenueId = useMemo(
+    () =>
+      relatedVenues.find((venue) => room?.url?.endsWith(`/${venue.id}`))?.id,
+    [relatedVenues, room?.url]
+  );
+
+  const {
+    loading: isLoadingRoomVenue,
+    error: roomVenueError,
+    value: roomVenue,
+  } = useAsync(async () => {
+    if (!roomVenueId) return;
+
+    return await fetchVenue(roomVenueId);
+  }, [roomVenueId]);
+
+  console.log(roomVenue);
+
+  const defaultValues = useMemo(
+    () => ({
+      room: {
+        title: room.title ?? "",
+        url: room.url ?? "",
+        subtitle: room.subtitle ?? "",
+        about: room.about ?? "",
+        template: room.template ?? undefined,
+        image_url: room.image_url ?? "",
+      },
+      venue: {
+        mapBackgroundImage: roomVenue?.mapBackgroundImageUrl ?? "",
+        zoomUrl: roomVenue?.zoomUrl ?? "",
+        iframeUrl: roomVenue?.iframeUrl ?? "",
+        showGrid: roomVenue?.showGrid ?? false,
+        showReactions: roomVenue?.showReactions ?? false,
+        showShoutouts: roomVenue?.showShoutouts ?? false,
+        auditoriumColumns: roomVenue?.auditoriumColumns ?? 0,
+        auditoriumRows: roomVenue?.auditoriumRows ?? 0,
+        columns: roomVenue?.columns ?? 0,
+      },
+    }),
+    [
+      room.about,
+      room.image_url,
+      room.subtitle,
+      room.template,
+      room.title,
+      room.url,
+      roomVenue?.auditoriumColumns,
+      roomVenue?.auditoriumRows,
+      roomVenue?.columns,
+      roomVenue?.iframeUrl,
+      roomVenue?.mapBackgroundImageUrl,
+      roomVenue?.showGrid,
+      roomVenue?.showReactions,
+      roomVenue?.showShoutouts,
+      roomVenue?.zoomUrl,
+    ]
+  );
+
+  const { register, handleSubmit, setValue, watch, reset, errors } = useForm({
+    reValidateMode: "onChange",
+    validationSchema: roomEditSchema,
+    defaultValues,
+  });
+
+  useEffect(() => reset(defaultValues), [defaultValues, reset]);
+
+  const values = watch("room");
+  const venueValues = watch("venue");
+
+  console.log(venueValues);
+
+  const changeRoomImageUrl = useCallback(
+    (val: string) => {
+      setValue("room.image_url", val, false);
+    },
+    [setValue]
+  );
+
+  const changeBackgroundImageUrl = useCallback(
+    (val: string) => {
+      setValue("venue.mapBackgroundImage", val, false);
+    },
+    [setValue]
+  );
+
+  const updateVenueRoom = useCallback(async () => {
+    if (!user || !roomVenueId) return;
+    await updateVenueNG(
+      {
+        id: roomVenueId,
+        ...venueValues,
+      },
+      user
+    );
+  }, [roomVenueId, user, venueValues]);
+
+  const [{ loading: isUpdating }, updateSelectedRoom] = useAsyncFn(async () => {
+    if (!user || !venueId) return;
+
+    const roomData: RoomInput = {
+      ...(room as RoomInput),
+      ...(updatedRoom as RoomInput),
+      ...values,
+    };
+
+    await upsertRoom(roomData, venueId, user, roomIndex);
+    console.log(room.template);
```

Comment:
let's remove console

Reply:
Fixed in https://github.com/sparkletown/sparkle/pull/2312/commits/cf45b3f8b266cc84df543b4a8c34438e9b9f6938